numpy/core/src/multiarray/item_selection.c

```diff
@@ -2154,20 +2218,25 @@ count_boolean_trues(int ndim, char *data, npy_intp const *ashape, npy_intp const
     }
 
     NPY_BEGIN_THREADS_THRESHOLDED(shape[0]);
-
     /* Special case for contiguous inner loop */
     if (strides[0] == 1) {
         NPY_RAW_ITER_START(idim, ndim, coord, shape) {
             /* Process the innermost dimension */
             const char *d = data;
             const char *e = data + shape[0];
+#if NPY_SIMD
+            npy_uintp stride = shape[0] & -npyv_nlanes_u8;
```

Comment:
Are there any guarantees about the relationship between `npyv_nlanes_u8` and `npyv_nlanes_u16`? I think the current code only works if the latter is a divisor of the former.

Reply:
in the non-dispatching mode, [Only one instruction set](https://github.com/numpy/numpy/blob/91d9bbebc159ada6cccd0e1fcec1926e26b19c6f/numpy/core/src/common/simd/simd.h#L30) will enabled, `npyv_nlanes_u8 `is twice as many as `npyv_nlanes_u16` under any circumstances.